### PR TITLE
HDDS-11194. OM missing audit log for upgrade prepare, cancel and finalize

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?

Audit log is added for upgrade prepare, cancel and finalize as this was missing. Since this is user request, this also needs logged.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-11194

## How was this patch tested?

- added testcase for audit log generation

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMPrepareRequest.java [hunk 1]
```diff
@@ -149,6 +157,8 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
       }
     }
 
+    auditLog(auditLogger, buildAuditMessage(OMAction.UPGRADE_PREPARE,
```
Comment (devmadhuu) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMPrepareRequest.java:160]: Can we add `transactionLogIndex`  and `OMNodeId` in auditMap as an additional info for audit message ?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java [hunk 2]
```diff
@@ -103,10 +110,13 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
           ozoneManager.getVersionManager().getMetadataLayoutVersion());
       LOG.trace("Returning response: {}", response);
     } catch (IOException e) {
+      exception = e;
       response = new OMFinalizeUpgradeResponse(
           createErrorOMResponse(responseBuilder, e), -1);
     }
 
+    auditLog(auditLogger, buildAuditMessage(OMAction.UPGRADE_FINALIZE,
+        new HashMap<>(), exception, userInfo));
```
Comment (devmadhuu) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java:119]: Can we add `termIndex` in auditMap as an additional info for audit message ?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMCancelPrepareRequest.java [hunk 3]
```diff
@@ -82,12 +89,15 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
...
     } catch (IOException e) {
+      exception = e;
       LOG.error("Cancel Prepare Request apply failed in {}. ",
           ozoneManager.getOMNodeId(), e);
       response = new OMPrepareResponse(
           createErrorOMResponse(responseBuilder, e));
     }
 
+    auditLog(auditLogger, buildAuditMessage(OMAction.UPGRADE_CANCEL,
+        new HashMap<>(), exception, userInfo));
```
Comment (devmadhuu) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMCancelPrepareRequest.java:100]: Can we add `OMNodeId` and `termIndex` in `auditMap` as an additional info for audit message ?

**Review by devmadhuu — COMMENTED** [review]
Thanks @sumitagrawl for working on this patch. Changes looks good. Just a few minor comments.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMCancelPrepareRequest.java (hunk 3 above)
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMCancelPrepareRequest.java:100]: TermIndex will be handled as part of "HDDS-10658. Add Object ID and Update ID to OM audit log messages" when this is merged.
OMNodeId is not required as audit-log is per node itself. There is no need for same.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java (hunk 2 above)
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java:119]: TermIndex will be handled as part of "[HDDS-10658](https://issues.apache.org/jira/browse/HDDS-10658). Add Object ID and Update ID to OM audit log messages" when this is merged.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMPrepareRequest.java (hunk 1 above)
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMPrepareRequest.java:160]: TermIndex will be handled as part of "[HDDS-10658](https://issues.apache.org/jira/browse/HDDS-10658). Add Object ID and Update ID to OM audit log messages" when this is merged implicitly.
OMNodeId is not required as audit-log is per node itself. There is no need for same.

**Review by devmadhuu — APPROVED** [review]
Thanks @sumitagrawl for the clarification. LGTM +1

**Comment by devmadhuu** [comment]
Thanks @sumitagrawl for the patch.

**Pull request merged**